plotter/step.go

```diff
@@ -0,0 +1,173 @@
+// Copyright ©2015 The gonum Authors. All rights reserved.
+// Use of this source code is governed by a BSD-style
+// license that can be found in the LICENSE file.
+
+package plotter
+
+import (
+	"image/color"
+	"math"
+
+	"gonum.org/v1/plot"
+	"gonum.org/v1/plot/vg"
+	"gonum.org/v1/plot/vg/draw"
+)
+
+// StepType specifies a form of a connection of two consecutive points.
+type StepType int
+
+const (
+	// StepTypePre means that two consecutive points are connected by two lines in the following order: vertical, horizontal.
+	StepTypePre StepType = iota
+
+	// StepTypeMid means that two consecutive points are connected by three lines in the following order: horizontal, vertical, horizontal.
+	// Vertical line is placed in the middle of the interval.
+	StepTypeMid
+
+	// StepTypePost means that two consecutive points are connected by two lines in the following order: horizontal, vertical.
+	StepTypePost
+)
+
+// Step implements the Plotter interface, drawing a stepped line.
+type Step struct {
+	// XYs is a copy of the points for this line.
+	XYs
+
+	// StepStyle is the type of step line
+	StepType StepType
+
+	// LineStyle is the style of the line connecting the points.
+	LineStyle *draw.LineStyle
+
+	// ShadeColor is the color of the shaded area.
```

Comment:
we should probably document that when left alone it wouldn't fill anything.

Reply:
Sure!

Reply:
Done